test/com/facebook/buck/jvm/java/JavaOrderingIntegrationTest.java

```diff
@@ -0,0 +1,135 @@
+      "1-2.txt:1",
+      "1-3.txt:1",
+      "1-4.txt:1",
+      "1-5.txt:1",
+      "2-3.txt:2",
+      "2-4.txt:2",
+      "2-5.txt:2",
+      "3-4.txt:3",
+      "3-5.txt:3",
+      "4-5.txt:4");
+
+  private static final String TEST_OUTPUT_MARKER = "===+===";
+
+  @Rule
+  public TemporaryPaths tmp = new TemporaryPaths();
+
+  @Test
+  public void testJavaBinaryOrdering() throws IOException {
+    ProjectWorkspace workspace = setUpProjectWorkspace();
+    Path path = workspace.buildAndReturnOutput("//buildables:bin");
+    try (ZipFile zip = new ZipFile(path.toString())) {
+      Assert.assertEquals("Expecting META-INF + 10 x-y.txt resources", 12, zip.stream().count());
+      StringBuilder resources = new StringBuilder();
+      for (int i = 1; i < 6; i++ ) {
+        for (int j = i + 1; j < 6; j++ ) {
+          String resource = i + "-" + j + ".txt";
+          InputStream inputStream = zip.getInputStream(zip.getEntry(resource));
+          Integer content =
+              Integer.parseInt(String.valueOf((char)(new InputStreamReader(inputStream)).read()));
+          inputStream.close();
+          resources.append(resource + ":" + content + ",");
+        }
+      }
+      resources.deleteCharAt(resources.length() - 1);
+      Assert.assertEquals(
+          "java_binary is expected to resolve conflicts in dependency in guaranteed order:",
+          JAVA_RESOLUTION_ORDER,
+          resources.toString());
+    }
+  }
+
+  @Test
+  public void testJavaTestOrdering() throws IOException {
+    ProjectWorkspace workspace = setUpProjectWorkspace();
+    ProcessResult result = workspace.runBuckCommand("test", "//buildables:test");
+    result.assertTestFailure();
+    String stderr = result.getStderr();
+    Assert.assertTrue(
+        "Expecting test to contain payload start marker, instead failed with " + stderr,
+        stderr.contains(TEST_OUTPUT_MARKER));
+    String markedOutput =
+        stderr.substring(stderr.indexOf(TEST_OUTPUT_MARKER) + TEST_OUTPUT_MARKER.length());
+    Assert.assertTrue(
+        "Expecting test to contain payload end marker, instead marked suffix is " + markedOutput,
+        stderr.contains(TEST_OUTPUT_MARKER));
+    markedOutput = markedOutput.substring(0, markedOutput.indexOf(TEST_OUTPUT_MARKER));
+    Assert.assertEquals(JAVA_RESOLUTION_ORDER, markedOutput);
+  }
+
+  @Test
+  public void testJavaBinaryClasspath() throws IOException {
+    ProjectWorkspace workspace = setUpProjectWorkspace();
+    testClassPathAudit(workspace, 5, "//buildables:bin");
+  }
+
+  @Test
+  public void testJavaBTestClasspath() throws IOException {
+    ProjectWorkspace workspace = setUpProjectWorkspace();
+    testClassPathAudit(workspace, 8, "//buildables:test");
+  }
+
+  private void testClassPathAudit(
+      ProjectWorkspace workspace, int expectedClassPathLength, String target) throws IOException {
+    ProcessResult result = workspace.runBuckCommand("audit", "classpath", target);
```

Comment:
This should be in `AuditClasspathCommandTest`.

Reply:
I intentionally had all of these in the same test so it is clear that these are related and have to be in the same ordering. I can split it across specific tests if you still think it is more clean.